common/io/src/unmarshal.rs

```diff
@@ -113,3 +113,13 @@ impl Unmarshal<bool> for bool {
         scratch[0] != 0
     }
 }
+
+impl Unmarshal<char> for char {
+    fn unmarshal(scratch: &[u8]) -> char {
+        let bits = u32::from(scratch[0])
+            | u32::from(scratch[1]) << 8
+            | u32::from(scratch[2]) << 16
+            | u32::from(scratch[3]) << 24;
+        Self::from_u32(bits).unwrap_or(' ')
```

Comment:
What does `unwrap_or(' ')` mean?
Why is it a space by default?

Reply:
As above said, `char` is just `u32` but not all `u32` is a valid `char`, return space when encounter invalid char seems fine here.

Reply:
If it is possible to fail, then it should be better to return `Result` or `panic`.
Otherwise, it just hides a bug.